### ISSUE TITLE
word emb and char emb doesn't match error

### ISSUE THREAD

**Issue by aopusd** [IsaacChanghau/neural_sequence_labeling#1]
tensorflow.python.framework.errors_impl.InvalidArgumentError: All dimensions except 2 must match. Input 1 has shape [20 40 200] and doesn't match input 0 with shape [20 0 300].
how to solve this error

**Comment by IsaacChanghau** [comment]
Hi aopusd, fixed the bugs you mentioned, and tested on my side, works properly, pls check the last updates. Thank you

**Comment by IsaacChanghau** [comment]
@aopusd Hi aopusd, fixed the bugs you mentioned, and tested on my side, works properly, pls check the last updates. Thank you
```
Start training...
Build model... done.
Epoch  1/30:
  1/703 [..............................] - ETA: 784s - train loss: 46.9639
  2/703 [..............................] - ETA: 702s - train loss: 31.7004
  3/703 [..............................] - ETA: 655s - train loss: 25.5127
  4/703 [..............................] - ETA: 539s - train loss: 20.6674
  5/703 [..............................] - ETA: 510s - train loss: 18.2604
  6/703 [..............................] - ETA: 516s - train loss: 17.1267
```

**Comment by aopusd** [comment]
非常感谢您百忙之中抽空回复，太感谢了！！！






------------------ 原始邮件 ------------------
发件人: "ZHANG HAO"<notifications@github.com>;
发送时间: 2018年3月5日(星期一) 中午11:20
收件人: "IsaacChanghau/neural_sequence_labeling"<neural_sequence_labeling@noreply.github.com>;
抄送: "奥普"<137219067@qq.com>; "Mention"<mention@noreply.github.com>; 
主题: Re: [IsaacChanghau/neural_sequence_labeling] word emb and char embdoesn't match error (#1)




Closed #1.

**Comment by aopusd** [comment]
Dear friend:
   when i use this model to deal with chinese chunking ,the has error,like below:
tensorflow.python.framework.errors_impl.InvalidArgumentError: indices[7,11,0] = 5981 is not in [0, 5980)
[[Node: char_rep_method/char_embeddings = Gather[Tindices=DT_INT32, Tparams=DT_FLOAT, _class=["loc:@char_rep_method/_char_embeddings"], validate_indices=true, _device="/job:localhost/replica:0/task:0/device:CPU:0"](char_rep_method/_char_embeddings/read, _arg_char_ids_0_0)]]
Caused by op 'char_rep_method/char_embeddings', defined at:







i think there is error in below code :
                if self.cfg.char_rep_method == 'rnn':
                    #char_embeddings = tf.reshape(char_embeddings, shape=[s[0] * s[1], s[-2], self.cfg.char_dim])
                    char_embeddings = tf.reshape(char_embeddings, shape=[s[0] , s[-2]* s[1], self.cfg.char_dim])
                    #embedded_chars为输入的tensor，[batch_szie, max_time, depth]。batch_size为模型当中batch的大小，应用在文本中时，
                    #max_time可以为句子的长度（一般以最长的句子为准，短句需要做padding），depth为输入句子词向量的维度。
                    word_lengths = tf.reshape(self.word_lengths, shape=[s[0] * s[1]])
                    char_bi_rnn = BiRNN(self.cfg.num_units_char, scope='char_rnn')

can i become one of the code contributor to fix some problems,thanks！
 
------------------ 原始邮件 ------------------
发件人: "ZHANG HAO"<notifications@github.com>;
发送时间: 2018年3月5日(星期一) 中午11:20
收件人: "IsaacChanghau/neural_sequence_labeling"<neural_sequence_labeling@noreply.github.com>;
抄送: "奥普"<137219067@qq.com>; "Mention"<mention@noreply.github.com>; 
主题: Re: [IsaacChanghau/neural_sequence_labeling] word emb and char embdoesn't match error (#1)




Closed #1.

**Comment by IsaacChanghau** [comment]
Hi,

I am not quite sure the errors you mentioned, can you give more details, and another thing is you are using the codes to tackle Chinese chunking problem, I am not sure if the Chinese data is suitable for the dat reprocessing in my codes. And if you want, you can directly fork my repo and make some changes.

Best regards,

ZHANG HAO
2680819187@qq.com
